File: packages/agentaid-py/src/agentaid/otel/conventions.py
from __future__ import annotations

from enum import StrEnum


class GenAI(StrEnum):
    SYSTEM = "gen_ai.system"
    REQUEST_MODEL = "gen_ai.request.model"
    RESPONSE_MODEL = "gen_ai.response.model"
    USAGE_INPUT_TOKENS = "gen_ai.usage.input_tokens"
    USAGE_OUTPUT_TOKENS = "gen_ai.usage.output_tokens"
    OPERATION_NAME = "gen_ai.operation.name"
    TOOL_NAME = "gen_ai.tool.name"
    TOOL_CALL_ID = "gen_ai.tool.call.id"

class AgentAid(StrEnum):
    """AgentAid-specific extensions."""
    RUN_ID = "agentaid.run_id"
    ROLE = "agentaid.role"
    PROMPT_SHA = "agentaid.prompt_sha"
    AGENT_NAME = "agentaid.agent_name"
    EVAL_RESULT = "agentaid.eval_result"
    INPUT = "agentaid.input"
    OUTPUT = "agentaid.output"

File: packages/agentaid-ts/src/otel/conventions.ts
export const GenAI = {
  System: "gen_ai.system",
  RequestModel: "gen_ai.request.model",
  ResponseModel: "gen_ai.response.model",
  UsageInputTokens: "gen_ai.usage.input_tokens",
  UsageOutputTokens: "gen_ai.usage.output_tokens",
  OperationName: "gen_ai.operation.name",
  ToolName: "gen_ai.tool.name",
  ToolCallId: "gen_ai.tool.call.id",
} as const;

export const AgentAid = {
  RunId: "agentaid.run_id",
  Role: "agentaid.role",
  PromptSha: "agentaid.prompt_sha",
  AgentName: "agentaid.agent_name",
  EvalResult: "agentaid.eval_result",
  Input: "agentaid.input",
  Output: "agentaid.output",
} as const;
